packages/edit-navigation/src/store/utils.js

```diff
@@ -172,3 +187,85 @@ export function computeCustomizedAttribute(
 		return omit( menuItemsByClientId[ block.clientId ] || {}, '_links' );
 	}
 }
+
+/**
+ * Maps a given block attribute to its equivalent menu item field.
+ *
+ * @param {string} blockAttributeName the name of the block attribute to be mapped.
+ * @param {string} blockAttributeValue the value of the block attribute to be mapped.
+ * @return {Object} block attribute mapped to correct menu item.
+ */
+export const mapBlockAttributeToMenuItemField = function mapBlockAttributeToMenuItemField(
+	blockAttributeName,
+	blockAttributeValue
+) {
+	const MAPPING = {
+		id: 'object_id',
+		type: 'object',
+		// `nav_menu_item` may be one of:
+		// 1. `post_type`,
+		// 2. `post_type_archive`,
+		// 3. `taxonomy`,
+		// 4. `custom`.
+		kind: {
+			attr: 'type', // the equivalent on `nav_menu_item` is "type" but it is stored with underscores whereas the block attr uses hyphens.
+			mapper: () => blockAttributeValue.replace( '-', '_' ),
```

Comment:
🤔 from the usage above, it might make sense to write this as an object to object transform vs per field. I think doing this per field adds some indirection and makes it a bit harder to read. A good exercise would be to port over the other items like title to see if what pattern we like better. Readability is subjective of course, so folks are welcome to chime in too

Note that data is also likely to be missing or malformed. A json schema might make sense in the transform function, though we'd prefer a decent attempt at fixing the data vs throwing an error (say if `kind` does not match the enum of values).

```jsx
const blockAttributesToMenuItem = ( { label, url, description, id, kind } ) => { 
    return {
        title: label,
        object_id: id,
        type: kind.replace( '-', '_' ),
        // and so on...
    }
}
```

Reply:
I've now changed this to use a model similar to that which you suggest here.

I went the other way mainly to avoid having to introduce too many changes into this PR but it ended up introducing unwanted complexity and indirection (as you correctly pointed out).